include/vapor/FlowParams.h

```diff
@@ -0,0 +1,157 @@
+#ifndef FLOWPARAMS_H
+#define FLOWPARAMS_H
+
+#include <vapor/RenderParams.h>
+#include <vapor/DataMgr.h>
+#include <vector>
+#include <utility>
+
+namespace VAPoR 
+{
+
+//
+// These two enums are used across params, GUI, and renderer.
+// Note: use static_cast to cast between them and int types.
+//
+enum class FlowSeedMode : int   
+{   
+    UNIFORM     = 0,
+    RANDOM      = 1,
+    RANDOM_BIAS = 2,
+    LIST        = 3 
+};  
+enum class FlowDir : int
+{   
+    FORWARD     = 0,
+    BACKWARD    = 1,
+    BI_DIR      = 2 
+}; 
+
+
+
+class PARAMS_API FlowParams : public RenderParams 
+{
+public:
+
+    // Constructors
+    FlowParams( DataMgr*                 dataManager, 
+                ParamsBase::StateSave*   stateSave );
+    FlowParams( DataMgr*                 dataManager, 
+                ParamsBase::StateSave*   stateSave, 
+                XmlNode*                 xmlNode );
+
+    virtual ~FlowParams();
+
+    //
+    // (Pure virtual methods from RenderParams)
+    //
+    virtual bool IsOpaque() const override
+    { 
+        return false; 
+    }
+    virtual bool usingVariable(const std::string& varname) override
+    {
+        return false;
+    }
+
+    static std::string GetClassType() 
+    {
+        return ("FlowParams");
+    }
+
+    // True  == Steady; False == Unteady
+    void   SetIsSteady( bool steady );
+    bool   GetIsSteady() const;
+
+    double GetVelocityMultiplier() const;
+    void   SetVelocityMultiplier( double );
+
+    long   GetSteadyNumOfSteps() const;
+    void   SetSteadyNumOfSteps( long );
+
+    int    GetSeedGenMode() const;
+    void   SetSeedGenMode( int );
+
+    void   SetNeedFlowlineOutput( bool );
+    bool   GetNeedFlowlineOutput( ) const;
+
+    int    GetFlowDirection() const;
+    void   SetFlowDirection( int );
+
+    std::string GetSeedInputFilename() const;
+    void        SetSeedInputFilename( const std::string& ) ;
+
+    std::string GetFlowlineOutputFilename() const;
+    void        SetFlowlineOutputFilename( const std::string& ) ;
+
+    std::vector<bool>   GetPeriodic() const;
+    void                SetPeriodic( const std::vector<bool>& );
+
+    /* 
+     * 6 values to represent a rake in this particular order: 
+     * xmin, xmax, ymin, ymax, zmin, zmax 
+     * If the rake wasn't set by users, it returns a vector containing nans.
+     */
+    std::vector<float>  GetRake() const;
```

Comment:
Everything else in Vapor uses the Params Box class to store regions. Please use that instead of the proprietary method here.

Reply:
The old Params Box is hard wired to the params entry of the user-defined Geometry region. Rake needs to be wired to a different params entry, thus not be able to use ParamsBox.